libvast/src/system/start_command.cpp

```diff
@@ -118,9 +143,9 @@ caf::message start_command(const invocation& inv, caf::actor_system& sys) {
         return caf::make_message(hook_invocation.error());
       detail::merge_settings(inv.options, hook_invocation->options,
                              policy::merge_lists::yes);
-      auto result = run(*hook_invocation, sys, root_factory);
-      if (!result)
-        return caf::make_message(result.error());
+      auto runner = self->spawn(command_runner);
+      command_runners.push_back(runner);
+      self->send(runner, atom::run_v, *hook_invocation);
```

Comment:
Instead of maintaining the runners in a separate vector, would it make sense to use `request.then()` and to capture the runner in the response handling lambdas? That way you don't keep the actors alive for longer than you actually need to.

We should be using request here anyways for logging, I think, in case an error is returned.

Reply:
We can't use `then()` here because `self` is a blocking actor.

For now I've just logged the error in the handler itself, although I think the "proper" solution would be to introduce a `struct command_result { optional<caf::error> e; }` that we can receive in the `do_receive` loop below.

Reply:
Fine by me.